### ISSUE TITLE
gpinitsystem failed and please help

### ISSUE THREAD

**Issue by abekn** [greenplum-db/gpdb#1568]
OS: OSX Sierra 10.12.2
SOURCE: master at today
command:  gpinitsystem -c gpinitsystem_config -D

gpinitsystem failed with below message.

***

20170117:03:17:58:059706 gpstart:mdw:admin-[DEBUG]:-[worker1] got cmd: $GPHOME/sbin/gpsegstart.py -C en_US.utf-8:en_US.utf-8:en_US.utf-8 -M mirrorless -V 'postgres (Greenplum Database) 5.0.0 build dev' -n 4 --era 9ed8e5a79495d432_170117031755 -t 600 -v -D '2|0|p|p|s|u|sdw1.local|sdw1-in|40000|None|/data/primary/gpseg0||' -D '3|1|p|p|s|u|sdw1.local|sdw1-in|40001|None|/data/primary/gpseg1||' -D '4|2|p|p|s|u|sdw1.local|sdw1-in|40002|None|/data/primary/gpseg2||' -D '5|3|p|p|s|u|sdw1.local|sdw1-in|40003|None|/data/primary/gpseg3||' -l '/Users/admin/gpAdminLogs'
20170117:03:17:58:059706 gpstart:mdw:admin-[DEBUG]:-[worker1] finished cmd: remote segment starts on host 'sdw1.local' cmdStr='ssh -o 'StrictHostKeyChecking no' sdw1-in ". /usr/local/gpdb/greenplum_path.sh; $GPHOME/sbin/gpsegstart.py -C en_US.utf-8:en_US.utf-8:en_US.utf-8 -M mirrorless -V 'postgres (Greenplum Database) 5.0.0 build dev' -n 4 --era 9ed8e5a79495d432_170117031755 -t 600 -v -D '2|0|p|p|s|u|sdw1.local|sdw1-in|40000|None|/data/primary/gpseg0||' -D '3|1|p|p|s|u|sdw1.local|sdw1-in|40001|None|/data/primary/gpseg1||' -D '4|2|p|p|s|u|sdw1.local|sdw1-in|40002|None|/data/primary/gpseg2||' -D '5|3|p|p|s|u|sdw1.local|sdw1-in|40003|None|/data/primary/gpseg3||' -l '/Users/admin/gpAdminLogs'"'  had result: cmd had rc=1 completed=True halted=False
  stdout=''
  stderr='Error: unable to import module: No module named psutil
'
. 
20170117:03:17:59:059706 gpstart:mdw:admin-[INFO]:-Process results...
20170117:03:17:59:059706 gpstart:mdw:admin-[ERROR]:-gpstart failed.  exiting...
Traceback (most recent call last):
  File "/usr/local/gpdb/lib/python/gppylib/mainUtils.py", line 281, in simple_main_locked
    exitCode = commandObject.run()
  File "/usr/local/gpdb/bin/gpstart", line 177, in run
    success=self._start(segmentsToStart, invalidSegments, inactiveSegments)
  File "/usr/local/gpdb/bin/gpstart", line 461, in _start
    segmentStartResult = segmentStartOp.startSegments(self.gparray, segmentsToStart, startMode, self.era)
  File "/usr/local/gpdb/lib/python/gppylib/operations/startSegments.py", line 150, in startSegments
    assert totalToAttempt == len(result.getFailedSegmentObjs()) + len(result.getSuccessfulSegments())
AssertionError
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-WorkerPool haltWork()
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-[worker0] haltWork
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-[worker1] haltWork
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-[worker2] got a halt cmd
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-[worker2] haltWork
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-[worker3] got a halt cmd
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-[worker4] got a halt cmd
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-[worker3] haltWork
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-[worker4] haltWork
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-[worker0] got a halt cmd
20170117:03:17:59:059706 gpstart:mdw:admin-[DEBUG]:-[worker1] got a halt cmd
20170117:03:17:59:044047 gpinitsystem:mdw:admin-[WARN]:
20170117:03:17:59:044047 gpinitsystem:mdw:admin-[WARN]:-Failed to start Greenplum instance; review gpstart output to

***

There is "No module named psutil" in error.
But "psutil" is  already installed.

$ pip freeze
cffi==1.9.1
cryptography==1.7.1
enum34==1.1.6
idna==2.2
ipaddress==1.0.18
lockfile==0.12.2
paramiko==2.1.1
psutil==5.0.1
pyasn1==0.1.9
pycparser==2.17
six==1.10.0

**Comment by Chibin** [comment]
Thanks for the report.
We will put this into our backlog and look into it.

**Comment by jimmyyih** [comment]
```
cmdStr='ssh -o 'StrictHostKeyChecking no' sdw1-in ". /usr/local/gpdb/greenplum_path.sh; $GPHOME/sbin/gpsegstart.py
```

Can you give us the content of `/usr/local/gpdb/greenplum_path.sh` on both mdw and sdw1 hosts?  Inside greenplum_path.sh should have a PYTHONPATH, PATH, and DYLD_LIBRARY_PATH that could be incorrect.  Also can you verify for us that the greenplum_path.sh files are the same on both hosts and that sdw1 host also has those Python libraries installed (it could be that you configured one server but not the other).

You put the OS as OSX, which is kinda interesting to have an OSX cluster... I'm kinda jealous.

**Comment by funnyxj** [comment]
Error: unable to import module: No module named psutil
pip install psutil

**Comment by khuddlefish** [comment]
Are you using brew?
We were hitting a similar issue and discovered that some environment variables were not getting sourced properly when commands were run through ssh.

This is what we were hitting:
```
➜  gpdemo git:(ddboost_test) ✗ echo $PATH
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/opt/fzf/bin
➜  gpdemo git:(ddboost_test) ✗ ssh `hostname` 'echo $PATH'
/usr/bin:/bin:/usr/sbin:/sbin
➜  gpdemo git:(ddboost_test) ✗ python -c 'import psutil'
➜  gpdemo git:(ddboost_test) ✗ echo $?
0
➜  gpdemo git:(ddboost_test) ✗ ssh `hostname` "python -c 'import psutil'"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named psutil
```

To resolve this issue, we modified our ssh configuration.
`echo "PATH=$PATH" >> ~/.ssh/environment #this assumes that you have /usr/local/bin already in the path.`
We also set a property `PermitUserEnvironment yes` in `/etc/ssh/sshd_config`

This allows ssh to use your path when running a command so it will be able to find the correct version of python.

**Comment by abekn** [comment]
Thank you for everyone.

I can have avoided  this issue.
This issue happened because of python path.
Sierra initially has python in "/use/bin" , but I added install python using BREW.
Then, Sierra has 2 python as PATH "/use/bin" and "/usr/local/bin" .
After that, Sierra generally used pyhton module  of "/usr/local/bin" .
Maybe, I installed  "psutil" module  into "/usr/local/bin" using  "/usr/local/bin/pip" command
But GreenplumDB tried to search pyhton module  to "/use/bin" generally  when gpinitsystem.
As result, GreenplumDB could not search "psutil" module even though Sierra had "psutil" module.
As workaround, I removed python from  "/usr/local/bin"  using BREW.

Thanks , and sorry for my mistake ,
